### PR TITLE
Handle datetimerange prefix modifiers

### PR DESCRIPTION
**Pull request by wgx998877**
GitHub issue https://github.com/Microsoft/Recognizers-Text/issues/329

Also added test cases for:

- Cortana can find us a time beginning of next week
- Sure, let's get a Skype end of next week
- Sure, let's get a Skype start of next week
- Cortana, find us a time end of March
- Cortana, please find us a time mid next week
- cortana can arrange us to meet mid March
- what about by mid summer?

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -46,15 +46,20 @@ public static class DateTimeDefinitions
...
 		public static readonly string BetweenRegex = $@"\b(between\s+)({DayRegex})\s*{RangeConnectorRegex}\s*({DayRegex})\s+{MonthSuffixRegex}((\s+|\s*,\s*){YearRegex})?\b";
 		public static readonly string MonthWithYear = $@"\b((?<month>April|Apr|August|Aug|December|Dec|February|Feb|January|Jan|July|Jul|June|Jun|March|Mar|May|November|Nov|October|Oct|September|Sep|Sept),?(\s+of)?\s+({YearRegex}|(?<order>next|last|this)\s+year))";
-		public static readonly string OneWordPeriodRegex = $@"\b((((the\s+)?month of\s+)?({RelativeRegex}\s+)?(?<month>April|Apr|August|Aug|December|Dec|February|Feb|January|Jan|July|Jul|June|Jun|March|Mar|May|November|Nov|October|Oct|September|Sep|Sept))|(month|year) to date|({RelativeRegex}\s+)?(weekend|week|month|year)(\s+{AfterNextSuffixRegex})?)\b";
+		public static readonly string OneWordPeriodRegex = $@"\b((((the\s+)?month of\s+)?({RelativeRegex}\s+)?(?<month>April|Apr|August|Aug|December|Dec|February|Feb|January|Jan|July|Jul|June|Jun|March|Mar|May|November|Nov|October|Oct|September|Sep|Sept))|(month|year) to date|({RelativeRegex}\s+)?((weekend|week|month|year))(\s+{AfterNextSuffixRegex})?)\b";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: Why do we need the extra ()?

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/IDateTimeParser.cs [hunk 2]
```diff
@@ -41,6 +41,10 @@ public static class TimeTypeConstants
         public const string afterMod = "after";
         public const string sinceMod = "since";
 
+        public const string earlyMod = "start";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/IDateTimeParser.cs]: Let's rename these to follow the same standard as the other constants in this file.
And let's rename the old START and END to RESOLVE_START and RESOLVE_END to make it clear they are not mod attributes.

Comment (wgx998877) [reply .NET/Microsoft.Recognizers.Text.DateTime/Parsers/IDateTimeParser.cs]: fixed

File: .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs (hunk 1 above)
Comment (wgx998877) [reply .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: removed

**Pull request merged**